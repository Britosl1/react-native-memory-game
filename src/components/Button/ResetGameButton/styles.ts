import styled from 'styled-components/native';

export const ResetButtonContainer = styled.TouchableOpacity`
  border-radius: 6px;
  background-color: #7c5295;
  height: 50px;
  width: 50px;
  align-items: center;
  justify-content: center;
  margin-bottom: 25px;
`;
