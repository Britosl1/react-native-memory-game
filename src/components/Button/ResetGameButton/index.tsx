import React from 'react';
import {Image} from 'react-native';
import ResetIcon from '../../../images/reset-arrow.png';
import {ResetButtonContainer} from './styles';

interface IResetGameButtonProps {
  onPress: () => void;
}

const ResetGameButton: React.FC<IResetGameButtonProps> = ({onPress}) => {
  return (
    <ResetButtonContainer onPress={onPress}>
      <Image source={ResetIcon} />
    </ResetButtonContainer>
  );
};

export default ResetGameButton;
