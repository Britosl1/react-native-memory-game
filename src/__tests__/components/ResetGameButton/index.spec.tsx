import React from 'react';
import {render} from '@testing-library/react-native';
import ResetGameButton from '../../../components/Button/ResetGameButton';

describe('Snapshot test', () => {
  test('Reset Button component', () => {
    const {toJSON} = render(<ResetGameButton onPress={jest.fn()} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
