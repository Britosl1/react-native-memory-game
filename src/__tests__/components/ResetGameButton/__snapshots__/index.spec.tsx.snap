// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Reset Button component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7c5295",
      "borderRadius": 6,
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 25,
      "opacity": 1,
      "width": 50,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/reset-arrow.png",
      }
    }
  />
</View>
`;
